backend/src/packages/users/user.service.ts

```diff
@@ -95,8 +98,21 @@ class UserService implements Service {
     return Promise.resolve(null);
   }
 
-  public delete(): ReturnType<Service['delete']> {
-    return Promise.resolve(true);
+  public async delete(
+    id: number,
+  ): Promise<ReturnType<UserEntity['toObject']> | null> {
+    const deletedUser = await this.userRepository.findById(id);
+
+    if (!deletedUser) {
+      throw new UsersError({
+        status: HTTPCode.NOT_FOUND,
+        message: ExceptionMessage.USER_NOT_FOUND,
+      });
+    }
+
+    const isDeleted = await this.userRepository.delete(id);
+
+    return isDeleted ? deletedUser.toObject() : null;
```

Comment:
Can you also patch `deletedAt` date in repo only for `whereNull('deletedAt')`? Then you needn't check `deletedUser` and can throw error right after `isDeleted === false` and return non nullable object
Anyway you're throwing error in auth service, so there's no matter to divide this functionality

![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92298163/37be8a6e-0a0b-49f8-802a-f446b83aa29d)